File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/service/fetchoptions/AggregationServiceFetchOptions.js
define([ "require", "stjs", "as/dto/common/fetchoptions/FetchOptions", "as/dto/service/fetchoptions/AggregationServiceSortOptions" ], function(require, stjs, FetchOptions) {
  var AggregationServiceFetchOptions = function() {
  };
  stjs.extend(AggregationServiceFetchOptions, FetchOptions, [ FetchOptions ], function(constructor, prototype) {
    prototype['@type'] = 'as.dto.service.fetchoptions.AggregationServiceFetchOptions';
    constructor.serialVersionUID = 1;
    prototype.sort = null;
    prototype.sortBy = function() {
      if (this.sort == null) {
        var AggregationServiceSortOptions = require("as/dto/service/fetchoptions/AggregationServiceSortOptions");
        this.sort = new AggregationServiceSortOptions();
      }
      return this.sort;
    };
    prototype.getSortBy = function() {
      return this.sort;
    };
  }, {
    sort : "AggregationServiceSortOptions"
  });
  return AggregationServiceFetchOptions;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/service/fetchoptions/AggregationServiceFetchOptions.java
/*
 * Copyright 2018 ETH Zuerich, SIS
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.service.fetchoptions;

import java.io.Serializable;

import com.fasterxml.jackson.annotation.JsonProperty;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.FetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.FetchOptionsToStringBuilder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.SortOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.service.AggregationService;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author Franz-Josef Elmer
 */
@JsonObject("as.dto.service.fetchoptions.AggregationServiceFetchOptions")
public class AggregationServiceFetchOptions extends FetchOptions<AggregationService> implements Serializable
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private AggregationServiceSortOptions sort;

    @Override
    public SortOptions<AggregationService> sortBy()
    {
        if (sort == null)
        {
            sort = new AggregationServiceSortOptions();
        }
        return sort;
    }

    @Override
    public SortOptions<AggregationService> getSortBy()
    {
        return sort;
    }

    @Override
    protected FetchOptionsToStringBuilder getFetchOptionsStringBuilder()
    {
        FetchOptionsToStringBuilder f = new FetchOptionsToStringBuilder("AggregationService", this);
        return f;
    }

}